sdk/servicebus/azure-servicebus/tests/perf_tests/_test_base.py

```diff
@@ -5,124 +5,210 @@
 
 import uuid
 
-from azure_devtools.perfstress_tests import PerfStressTest, get_random_bytes
+from azure_devtools.perfstress_tests import PerfStressTest, get_random_bytes, BatchPerfTest
 
-from azure.servicebus import ServiceBusClient, ServiceBusReceiveMode, ServiceBusMessage
+from azure.servicebus import ServiceBusClient, ServiceBusReceiveMode, ServiceBusMessage, TransportType
 from azure.servicebus.aio import ServiceBusClient as AsyncServiceBusClient
 from azure.servicebus.aio.management import ServiceBusAdministrationClient
 
-MAX_QUEUE_SIZE = 40960
+MAX_QUEUE_SIZE=4096
 
-
-class _ServiceTest(PerfStressTest):
-    service_client = None
-    async_service_client = None
-
-    def __init__(self, arguments):
+class _ReceiveTest(PerfStressTest):
+    def __init__(self, arguments) -> None:
         super().__init__(arguments)
-
-        connection_string = self.get_from_env("AZURE_SERVICEBUS_CONNECTION_STRING")
-        if self.args.no_client_share:
-            self.service_client = ServiceBusClient.from_connection_string(connection_string)
-            self.async_service_client = AsyncServiceBusClient.from_connection_string(connection_string)
-        else:
-            if not _ServiceTest.service_client:
-                _ServiceTest.service_client = ServiceBusClient.from_connection_string(connection_string)
-                _ServiceTest.async_service_client = AsyncServiceBusClient.from_connection_string(connection_string)
-            self.service_client = _ServiceTest.service_client
-            self.async_service_client =_ServiceTest.async_service_client
-
-
-    async def close(self):
-        self.service_client.close()
-        await self.async_service_client.close()
+        self.connection_string=self.get_from_env("AZURE_SERVICEBUS_CONNECTION_STRING")
+        transport_type=TransportType.AmqpOverWebsocket if arguments.transport_type == 1 else TransportType.Amqp
+        mode=ServiceBusReceiveMode.PEEK_LOCK if arguments.peeklock else ServiceBusReceiveMode.RECEIVE_AND_DELETE
+        self.servicebus_client=ServiceBusClient.from_connection_string(
+            self.connection_string,
+            receive_mode=mode,
+            prefetch_count=self.args.num_messages,
+            max_wait_time=self.args.max_wait_time or None, 
+            transport_type=transport_type,
+            uamqqp_transport=arguments.transport_type,
```

Comment:
will these run if we don't have the switch yet b/c of uamqp_transport?

Reply:
should we add it in if not already there so the perftests are happy?

Reply:
hmm it should noop imo. Its passed in as a kwarg and right now shouldnt be used anywhere

Reply:
also there's a typo:
```suggestion
            uamqp_transport=arguments.transport_type,
```